modules/client/src/types.ts

```diff
@@ -420,18 +420,65 @@ export type DecodedInitializeFromParams = {
 
 export type PluginUpdateProposalValidity = {
   isValid: boolean;
-  causes: PluginUpdateProposalInValidityCause[];
+  actionErrorCauses: PluginUpdateProposalInValidityCause[][];
+  proposalSettingsErrorCauses: ProposalSettingsErrorCause[]
 };
 
-export enum PluginUpdateProposalInValidityCause {
+export enum ProposalSettingsErrorCause {
+  INVALID_ALLOW_FAILURE_MAP = "invalidAllowFailureMap",
   INVALID_ACTIONS = "invalidActions",
-  INVALID_GRANT_PERMISSION = "invalidGrantPermission",
-  INVALID_REVOKE_PERMISSION = "invalidRevokePermission",
+  PROPOSAL_NOT_FOUND = "proposalNotFound",
+}
+
+export enum PluginUpdateProposalInValidityCause {
+  // Grant UPDATE_PLUGIN_PERMISSION action
+  INVALID_GRANT_UPDATE_PERMISSION_WHO_ADDRESS =
+    "invalidGrantUpdatePermissionWhoAddress",
+  INVALID_GRANT_UPDATE_PERMISSION_WHERE_ADDRESS =
+    "invalidGrantUpdatePermissionWhereAddress",
+  INVALID_GRANT_UPDATE_PERMISSION_PERMISSION =
```

Comment:
The difference between
INVALID_GRANT_UPDATE_PERMISSION_PERMISSION
and
INVALID_GRANT_UPDATE_PERMISSION_PERMISSION_ID
is not so clear from the name alone.

Reply:
Please create a task to rename "permission" to "permissionName" or everywhere.